cpp/arcticdb/storage/s3/s3_storage-inl.hpp

```diff
@@ -49,6 +49,32 @@ static const size_t DELETE_OBJECTS_LIMIT = 1000;
 template<class It>
 using Range = folly::Range<It>;
 
+template<class Result, class KeyType>
+void handle_common_errors(const Aws::Utils::Outcome<Result, Aws::S3::S3Error>& outcome,
+        const char* op,
+        const KeyType& key) {
+    using namespace Aws::S3;
+    if (!outcome.IsSuccess()) {
+        auto& err = outcome.GetError();
+        auto type = err.GetErrorType();
+        if (type == S3Errors::NO_SUCH_KEY || type == S3Errors::RESOURCE_NOT_FOUND) {
+            return; // Caller need to handle these
+        }
+
+#define S3_ERROR_FMT " error while {} {}: S3Error#{} {}: {}", \
```

Comment:
No real need for this to be a macro is there? Just a normal function would be fine?

Reply:
+1